internal/log/slog.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+)
+
+// NewLoggerSlog returns a *slog.Logger that prints in the provided format at the
+// provided level with a UTC timestamp and the caller of the log entry.
+func NewLoggerSlog(c Config) (*slog.Logger, error) {
+	lvlOption, err := parseLevel(c.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	handler, err := getHandlerFromFormat(c.Format, slog.HandlerOptions{
+		Level:       lvlOption,
+		AddSource:   true,
+		ReplaceAttr: replaceSlogAttributes,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return slog.New(handler), nil
+}
+
+func replaceSlogAttributes(_ []string, a slog.Attr) slog.Attr {
```

Comment:
```suggestion
// replaceSlogAttributes replaces fields that were added by default by slog, but had different 
// formats or key names in github.com/go-kit/log. The operator was originally implemented with go-kit/log,
// so we use these replacements to make the migration smoother.
func replaceSlogAttributes(_ []string, a slog.Attr) slog.Attr {
```

Reply:
instead of commit your suggestion I took the opportunity to change everything https://github.com/prometheus-operator/prometheus-operator/pull/6743/commits/1978e2c56139ac0b6d2b64f3562a9f1aecff4668
in a single shot